src/contest/weglide/UploadIGCFile.cpp

```diff
@@ -0,0 +1,207 @@
+};
+
+static void
+UploadJsonInterpreter(const boost::json::value &json,
+                      UploadResponse& response) {
+  if (json.is_array() && (json.get_array().size() > 0)) {
+    auto flight_obj = json.get_array().at(0);
+    response.scoring_date = GetJsonString(flight_obj, "scoring_date").c_str();
+    response.flight_id = flight_obj.at("id").to_number<int64_t>();
+    response.registration = GetJsonString(flight_obj, "registration").c_str();
+    response.competition_id = GetJsonString(flight_obj, "competition_id").c_str();
+    auto user = flight_obj.at("user");
+    if (user.is_object()) {
+      auto user_obj = user.as_object();
+      response.user.id = user_obj.at("id").to_number<uint32_t>();
+      response.user.name = GetJsonString(user_obj, "name").c_str();
+    }
+    auto aircraft = flight_obj.at("aircraft");
+    if (aircraft.is_object()) {
+      auto aircraft_obj = aircraft.as_object();
+      response.aircraft.id = aircraft_obj.at("id").to_number<uint32_t>();
+      response.aircraft.name = GetJsonString(aircraft_obj, "name").c_str();
+      response.aircraft.kind = GetJsonString(aircraft_obj, "kind").c_str();
+      response.aircraft.sc_class = GetJsonString(aircraft_obj, "sc_class").c_str();
+    }
+  }
+  else {
+    throw std::runtime_error("The http response has a wrong JSON format!");
+  }
+}
+
+static void
+UploadSuccessDialog(const UploadResponse& response, const Path igc_path) {
+  // TODO: Create a real Dialog with fields
+  // With this Dialog insert the possibilty to update/patch the flight
+  // f.e. copilot in double seater, scoring class, short comment and so on
+  StaticString<0x1000> msg;
+  msg.Format(_("File upload '%s' was succesfull"),
+    igc_path.c_str());
+  msg.append(_T("!\n\n"));
+  msg.AppendFormat(_T("%-20s: %u\n"), _("flight_id"), response.flight_id);
+  msg.AppendFormat(_T("%-20s: %s\n"), _("scoring_date"),
+    response.scoring_date.c_str());
+  msg.AppendFormat(_T("%-20s: %s (%d)\n"), _("user"),
+    response.user.name.c_str(), response.user.id);
+  msg.AppendFormat(_T("%-20s: %s (%u)\n"), _("aircraft type"),
+    response.aircraft.name.c_str(), response.aircraft.id);
+  msg.AppendFormat(_T("%-20s: %s: %s "), _("aircraft"), _("reg"),
+    response.registration.c_str());
+  msg.AppendFormat(_T("%s: %s\n"), _("cid"),
+    response.competition_id.c_str());
+
+  ShowMessageBox(msg, _T("WeGlide Upload"), MB_OK | MB_ICONEXCLAMATION);
+}
+
+static Co::InvokeTask
+UpdateTask(Path igc_path, const WeGlideSettings& settings,
+  uint_least32_t glider_id, boost::json::value& json,
+  ProgressListener& progress) {
+  json = co_await UploadFlight(*Net::curl, settings, glider_id,
+    igc_path, progress);
+}
+
+bool
+UploadIGCFile(Path filepath) {
+  const WeGlideSettings& weglide_settings =
+  CommonInterface::GetComputerSettings().weglide;
+  NarrowString<0x200> url(weglide_settings.default_url);
+  url += "/igcfile";
+  auto igc_path = filepath.IsAbsolute() ? AllocatedPath(filepath) :
```

Comment:
Bad coding style - and the `LocalPath()` call is **still** there.

Reply:
removed this part, url not needed here
with LocalPath: I thought in this frontend function it is allowed? Nevertheless I moved this to the caller function.